packages/protocol/contracts/tokenvault/BridgedERC20V2.sol

```diff
@@ -41,7 +42,7 @@ contract BridgedERC20V2 is BridgedERC20, IERC20PermitUpgradeable, EIP712Upgradea
         external
         virtual
         override
-        initializer
+        reinitializer(2)
```

Comment:
I was aware of the issue with the previous V2, but i thought we are not upgrading the previous tokens but going forward we will be using this in the bridge -> this case it would have been ok.

now, you need 2 transactions then in the bridge when a new token is bridged. One init, and an another one which initializes the eip712, or am i missing something ?

Reply:
> now, you need 2 transactions then in the bridge when a new token is bridged. One init, and an another one which initializes the eip712, or am i missing something ?

For new deployment, we only need to call `init` once, once it is called, `init2` can NOT be called as version is 2 after `init` is called. But for upgraded contracts, they need to call `init2` to change version from 1 to 2.

Reply:
> For new deployment, we only need to call `init` once, once it is called, `init2` can NOT be called as version is 2 after `init` is called. But for upgraded contracts, they need to call `init2` to change version from 1 to 2.

Yes but init() is in normal BridgedERC20, no ? Does ir initializes the eip712 ? (From mobile and cannot see the code changes properly on small screen).

Reply:
Sorry, I don't understand the question...

Reply:
Sorry, i did not see properly on mobile an cannot type my question. But after reviewing this solution, i think it works for retrospecitve obviously + future deployments, so all good it seems.